### PR TITLE
feat(sdk-coin-celo): support eip1559 for celo

### PR DESCRIPTION
**Pull request by mullapudipruthvik**
Ticket: COIN-2046

TICKET: COIN-2046

Changes only to testnet

### CODE REVIEW

File: modules/sdk-coin-celo/test/unit/transactionBuilder/send.ts [hunk 1]
```diff
@@ -183,7 +213,7 @@ describe('Send transaction', function () {
         .key(key);
       const tx = await txBuilder.build();
       const txJson = tx.toJson();
-      should.equal(txJson.v, '0xaef3');
+      should.equal(txJson.v, '0x015e09');
```
Comment (mullapudipruthvik) [inline modules/sdk-coin-celo/test/unit/transactionBuilder/send.ts:216]: after the hardfork, `v` value should be 2*chainId + 35

File: modules/sdk-coin-celo/src/lib/transaction.ts [hunk 2]
```diff
@@ -1,21 +1 @@
...
-  /** @inheritdoc */
-  public static fromSerialized(
-    coinConfig: Readonly<CoinConfig>,
-    common: EthereumCommon,
-    serializedTx: string
-  ): Transaction {
-    return new Transaction(coinConfig, common, Utils.deserialize(serializedTx));
-  }
-}
+export { Transaction } from '@bitgo/sdk-coin-eth';
```
Comment (gianchandania) [inline modules/sdk-coin-celo/src/lib/transaction.ts:1]: I think Transaction can be used from abstract-eth

File: modules/sdk-coin-celo/src/lib/transaction.ts [hunk 3]
```diff
@@ -1,21 +1 @@
-import { BaseCoin as CoinConfig } from '@bitgo/statics';
```
Comment (gianchandania) [inline modules/sdk-coin-celo/src/lib/transaction.ts:1]: Wondering if because of this change, it needs to be marked as breaking change?

**Review by gianchandania — DISMISSED** [review]

File: modules/sdk-coin-celo/src/lib/testnetTransactionBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    super(_coinConfig);
+    this.transaction = new Transaction(_coinConfig, this._common);
+  }
+
+  /**
+   * Get the final v value. Final v is described in EIP-155.
+   *
+   * @protected for internal use when the enableFinalVField flag is true.
+   */
+  protected getFinalV(): string {
```
Comment (gianchandania) [inline modules/sdk-coin-celo/src/lib/testnetTransactionBuilder.ts:17]: This method is not needed for eip1559, right? Why are we adding this?

File: modules/sdk-coin-celo/src/lib/testnetTransactionBuilder.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    return ethUtil.addHexPrefix(this._common.chainIdBN().muln(2).addn(35).toString(16));
+  }
+
+  /** @inheritdoc */
+  protected fromImplementation(rawTransaction: string): Transaction {
```
Comment (gianchandania) [inline modules/sdk-coin-celo/src/lib/testnetTransactionBuilder.ts:22]: This method is almost similar in the celo transaction builder, what is the reason to add this here?

File: modules/sdk-coin-celo/src/lib/resources.ts [hunk 6]
```diff
@@ -4,25 +4,23 @@ import EthereumCommon from '@ethereumjs/common';
...
-export const testnetCommon = EthereumCommon.forCustomChain(
-  'mainnet', // It's a test net based on the main ethereum net
+export const testnetCommon = EthereumCommon.custom(
   {
     name: 'alfajores',
     networkId: (coins.get('tcelo').network as EthereumNetwork).chainId,
     chainId: (coins.get('tcelo').network as EthereumNetwork).chainId,
   },
-  'petersburg'
+  { hardfork: 'london' }
```
Comment (adarsh312) [inline modules/sdk-coin-celo/src/lib/resources.ts:13]: should this be 'petersburg' ? how are we determining this ?

**Review by gianchandania — APPROVED** [review]

**Pull request merged**